AppDB/appscale/datastore/scripts/ua_server_backup.py

```diff
@@ -0,0 +1,134 @@
+""" UA Server data backup. """
+
+import argparse
+import errno
+import logging
+import os
+import time
+
+from tornado import gen
+
+from appscale.common.constants import LOG_FORMAT
+from appscale.datastore import appscale_datastore
+from appscale.datastore.dbconstants import (
+  AppScaleDBConnectionError,
+  USERS_SCHEMA,
+  USERS_TABLE
+)
+from appscale.datastore.utils import tornado_synchronous
+
+# The location where the backup files will be stored.
+BACKUP_FILE_LOCATION = "/opt/appscale/backups/"
+
+
+@tornado_synchronous
+@gen.coroutine
+def get_table_sync(datastore, table_name, schema):
+  """ Gets data from datastore.
+
+  Args:
+    datastore: Datastore entity.
+    table_name: Table name in datastore.
+    schema: Table schema.
+  """
+  result = yield datastore.get_table(table_name, schema)
+  raise gen.Return(result)
+
+def reshape(array, step):
+  """ Reshapes array of size n to matrix with dimensions n/step by step.
+
+  Args:
+    array: List to reshape.
+    step: Number of elements in row after reshaping.
+  """
+  result = []
+  for i in range(0, len(array), step):
+    result.append(array[i:i+step])
+  return result
+
+def create_backup_dir(backup_dir):
+  """ Creates backup directory.
+
+  Args:
+    backup_dir: Backup directory name.
+  Returns:
+    True if directory was created or existed before, false otherwise.
+  """
+  logger = logging.getLogger(__name__)
+  try:
+    os.makedirs(backup_dir)
+    logger.info("Backup dir created: {0}".format(backup_dir))
+    return True
+  except OSError, os_error:
+    if os_error.errno == errno.EEXIST:
+      logger.warn("OSError: Backup directory already exists.")
+      return True
+    elif os_error.errno == errno.ENOSPC:
+      logger.error("OSError: No space left to create backup directory.")
+      logger.error(os_error.message)
+      return False
```

Comment:
As a general Python style tip, it's usually fine to let the exceptions propagate up unless there is a reason to silence or convert them. Returning `False` or `True` to indicate the success of the operation is not as conventional.

For example, this function can be simplified to
```
def create_backup_dir(backup_dir):
  try:
    os.makedirs(backup_dir)
  except OSError as os_error:
    if os_error.errno != errno.EEXIST:
      raise

  logger.info("Backup dir created: {0}".format(backup_dir))
```